PWGCF/DataModel/FemtoDerived.h

```diff
@@ -175,6 +179,108 @@ DECLARE_SOA_COLUMN(DecayVtxZ, decayVtxZ, float);     //! Z position of the decay
+DECLARE_SOA_COLUMN(MT, mT, float);
+DECLARE_SOA_COLUMN(CharmM, charmM, float);
+DECLARE_SOA_COLUMN(CharmPt, charmPt, float);
+DECLARE_SOA_COLUMN(Mult, mult, int);
+DECLARE_SOA_COLUMN(MultPercentile, multPercentile, float);
+DECLARE_SOA_COLUMN(PartPairSign, partPairSign, int8_t);
+DECLARE_SOA_COLUMN(ProcessType, processType, int64_t);
+DECLARE_SOA_DYNAMIC_COLUMN(M, m, //!
+                           [](float pt0, float phi0, float eta0, float pt1, float phi1, float eta1, float pt2, float phi2, float eta2, const std::array<double, 3>& m) -> float { return RecoDecay::m(std::array{std::array{pt0 * std::cos(phi0), pt0 * std::sin(phi0), pt0 * std::sinh(eta0)}, std::array{pt1 * std::cos(phi1), pt1 * std::sin(phi1), pt1 * std::sinh(eta1)}, std::array{pt2 * std::cos(phi2), pt2 * std::sin(phi2), pt2 * std::sinh(eta2)}}, m); });
+DECLARE_SOA_DYNAMIC_COLUMN(P, p, //!
+                           [](float pt0, float phi0, float eta0, float pt1, float phi1, float eta1, float pt2, float phi2, float eta2) -> float { return RecoDecay::p(RecoDecay::pVec(std::array{pt0 * std::cos(phi0), pt0 * std::sin(phi0), pt0 * std::sinh(eta0)}, std::array{pt1 * std::cos(phi1), pt1 * std::sin(phi1), pt1 * std::sinh(eta1)}, std::array{pt2 * std::cos(phi2), pt2 * std::sin(phi2), pt2 * std::sinh(eta2)})); });
+DECLARE_SOA_DYNAMIC_COLUMN(Pt, pt, //!
+                           [](float pt0, float phi0, float pt1, float phi1, float pt2, float phi2) -> float { return RecoDecay::pt(std::array{pt0 * std::cos(phi0), pt0 * std::sin(phi0)}, std::array{pt1 * std::cos(phi1), pt1 * std::sin(phi1)}, std::array{pt2 * std::cos(phi2), pt2 * std::sin(phi2)}); });
+DECLARE_SOA_DYNAMIC_COLUMN(Phi, phi, //!
+                           [](float pt0, float phi0, float eta0, float pt1, float phi1, float eta1, float pt2, float phi2, float eta2) -> float { return RecoDecay::phi(std::array{(pt0 * std::cos(phi0) + pt1 * std::cos(phi1) + pt2 * std::cos(phi2)), (pt0 * std::sin(phi0) + pt1 * std::sin(phi1) + pt2 * std::sin(phi2)), (pt0 * std::sinh(eta0) + pt1 * std::sinh(eta1) + pt2 * std::sinh(eta2))}); });
+DECLARE_SOA_DYNAMIC_COLUMN(Y, y, //!
+                           [](float pt0, float phi0, float eta0, float pt1, float phi1, float eta1, float pt2, float phi2, float eta2) -> float { return RecoDecay::y(std::array{(pt0 * std::cos(phi0) + pt1 * std::cos(phi1) + pt2 * std::cos(phi2)), (pt0 * std::sin(phi0) + pt1 * std::sin(phi1) + pt2 * std::sin(phi2)), (pt0 * std::sinh(eta0) + pt1 * std::sinh(eta1) + pt2 * std::sinh(eta2))}, o2::constants::physics::MassLambdaCPlus); });
+DECLARE_SOA_DYNAMIC_COLUMN(Eta, eta, //!
+                           [](float pt0, float phi0, float eta0, float pt1, float phi1, float eta1, float pt2, float phi2, float eta2) -> float { return RecoDecay::eta(std::array{(pt0 * std::cos(phi0) + pt1 * std::cos(phi1) + pt2 * std::cos(phi2)), (pt0 * std::sin(phi0) + pt1 * std::sin(phi1) + pt2 * std::sin(phi2)), (pt0 * std::sinh(eta0) + pt1 * std::sinh(eta1) + pt2 * std::sinh(eta2))}); });
```

Comment:
I suspect this would get much simplified if you could use the functionalities introduced in https://github.com/AliceO2Group/O2Physics/pull/6085